### ISSUE TITLE
Why is the output of sentence_transformers different from the average output of transformers with the same model?

### ISSUE THREAD

**Issue by dkobak** [UKPLab/sentence-transformers#1871]
Consider encoding some text using SBERT:
```Python
from sentence_transformers import SentenceTransformer
smodel = SentenceTransformer('sentence-transformers/all-mpnet-base-v2')

text = "This is not a pipe."

print(smodel.tokenize([text])['input_ids'])
print(smodel.encode([text])[0,:5])
```

The output is 
```Python
tensor([[   0, 2027, 2007, 2029, 1041, 8671, 1016,    2]])
[-0.02439323  0.01049526  0.00373949  0.03735257 -0.03130341]
```

Why is it different from
```Python
from transformers import AutoModel, AutoTokenizer
model = AutoModel.from_pretrained('sentence-transformers/all-mpnet-base-v2')
tokenizer = AutoTokenizer.from_pretrained('sentence-transformers/all-mpnet-base-v2')

input = tokenizer(text, return_tensors="pt")
print(input['input_ids'])
    
output = model(**input).last_hidden_state.squeeze().detach().numpy().mean(axis=0)
print(output[:5])
```
which prints
```Python
tensor([[   0, 2027, 2007, 2029, 1041, 8671, 1016,    2]])
[-0.06678095  0.02873271  0.01023754  0.10225952 -0.08569884]
```

So the tokenizer produces the same output, but the network output is different. I thought SBERT returns the average output embedding across all tokens; does it not?

**Comment by dkobak** [comment]
Okay, sorry, I figured it out: SBERT's output is normalized to have norm 1.
```Python
print((output/np.linalg.norm(output))[:5])
```
this gives the same result as SBERT. Closing this issue.

**Comment by Ravi2712** [comment]
@dkobak Wait! SBERT's default `normalize_embeddings` param is set to False. So, it shouldn't be the same according to your code. You don't set the argument to `True` for normalizing the final output. Can you confirm this? It would be great help, thanks.

**Comment by dkobak** [comment]
@Ravi2712 Hmm I checked and I do get normalized output without touching `normalize_embeddings`:

![Screenshot from 2023-05-25 08-54-54](https://github.com/UKPLab/sentence-transformers/assets/8970231/14776c65-43bc-4b6a-88dc-c4cc6a540124)

Why is that?
